src/org/loklak/api/search/SearchServlet.java

```diff
@@ -356,109 +363,11 @@ public void run() {
-                channel.setPubDate(new Date());
-                channel.setTitle("RSS feed for Twitter search for " + query);
-                channel.setDescription("");
-                channel.setLink("");
-                RSSFeed feed = new RSSFeed(tl.size());
-                feed.setChannel(channel);
-                try {
-                    for (TwitterTweet t : tl.getNextTweets(startRecord - 1, maximumRecords)) {
-                        UserEntry u = tl.getUser(t);
-                        RSSMessage m = new RSSMessage();
-                        m.setLink(t.getStatusIdUrl().toExternalForm());
-                        m.setAuthor(u.getName() + " @" + u.getScreenName());
-                        m.setTitle(u.getName() + " @" + u.getScreenName());
-                        m.setDescription(t.moreData.getText(shortlink_iflinkexceedslength, shortlink_urlstub, t.getText(), t.getLinks(), t.getPostId()).text);
-                        m.setPubDate(t.getCreatedAt());
-                        m.setGuid(t.getPostId());
-                        feed.addMessage(m);
-                    }
-                } catch (ConcurrentModificationException e) {
-                    // late incoming messages from concurrent peer retrieval may cause this
-                    // we silently do nothing here and return what we listed so far
-                }
-                String rss = feed.toString();
-                //System.out.println("feed has " + feed.size() + " entries");
-
-                // write xml
-                response.getOutputStream().write(UTF8.getBytes(rss));
-            } else if (txtExt) {
-                post.setResponse(response, "text/plain");
-                final StringBuilder buffer = new StringBuilder(1000);
-                try {
-                    for (TwitterTweet t : tl) {
-                        UserEntry u = tl.getUser(t);
-                        buffer.append(t.getCreatedAt()).append(" ")
-                                .append(u.getScreenName())
-                                .append(": ")
-                                .append(t.moreData.getText(
-                                        shortlink_iflinkexceedslength, shortlink_urlstub,
-                                        t.getText(), t.getLinks(), t.getPostId()).text)
-                                .append('\n');
-                    }
-                } catch (ConcurrentModificationException e) {
-                    // late incoming messages from concurrent peer retrieval may cause this
-                    // we silently do nothing here and return what we listed so far
-                }
-                response.getOutputStream().write(UTF8.getBytes(buffer.toString()));
-            }
+            dataOutput( response, dataArray, query, startRecord, maximumRecords, minified,
+                    callType, jsonp, post, tl, aggregations[0], hits, callback, shortlink_urlstub,
+                    shortlink_iflinkexceedslength, filter, metadata);
```

Comment:
yes, but then, it will be difficult to make further improvements and further increase the size of the `doGet` method. The `dataOutput` method has to be separated to make further changes like to enable output for different callTypes like xml, json and txt.